hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java

```diff
@@ -277,6 +219,15 @@ public void handleQuery(
         }
     }
 
+    private Query parseQuery(Bytes requestBuffer) {
+        try {
+            return queryParser.parseStrict(requestBuffer.toReadableSequentialData());
+        } catch (IOException e) {
+            // TODO there may be other types of errors here. Please cross check with ingest parsing
+            throw new StatusRuntimeException(Status.INVALID_ARGUMENT);
```

Comment:
```suggestion
            throw new StatusRuntimeException(Status.INVALID_ARGUMENT, e);
```

Reply:
`StatusRuntimeException` does not allow this.